share/models/registration.py

```diff
@@ -0,0 +1,41 @@
+from model_utils import Choices
+
+from django.conf import settings
+from django.db import models
+from django.utils.translation import ugettext as _
+
+
+class ProviderRegistration(models.Model):
+
+    STATUS = Choices(
+        (0, 'pending', _('pending')),
+        (1, 'accepted', _('accepted')),
+        (2, 'implemented', _('implemented')),
+        (3, 'rejected', _('rejected'))
+    )
+
+    status = models.IntegerField(choices=STATUS, default=STATUS.pending)
+    submitted_at = models.DateTimeField(auto_now_add=True)
+    submitted_by = models.ForeignKey(settings.AUTH_USER_MODEL)
+
+    contact_name = models.TextField()
+    contact_email = models.EmailField()
+    contact_affiliation = models.TextField()
+
+    meta_tos = models.BooleanField(default=False)
+    meta_rights = models.BooleanField(default=False)
+    meta_privacy = models.BooleanField(default=False)
+    meta_sharing = models.BooleanField(default=False)
+    meta_license = models.BooleanField(default=False)
+
+    direct_source = models.BooleanField(default=False)
+
+    source_name = models.TextField()
```

Comment:
We should probably put maximum lengths on all the textfields here, just in case. They can be fairly generous but draw the line somewhere. 🚒